### DISCUSSION TITLE
How can one transmit a custom fields like winds speed and heading over AIS?

### DISCUSSION THREAD

**Discussion by revolteur** [peterantypas/maiana#79] [Ideas]
I am wondering if it is possible to transmit a custom measurement like Wind speed and heading (T) using the AIS message standard. 
 
1) Does the standard support such functionality? 
I skimmed through the AIS messages standard but nothing obvious jumped at me. 
I also did not find a misc. message placeholder that could be used for such purpose. 

2) If the AIS standard does not support transmitting a misc. data point (ex: Wind, else), how would one go about adding such functionality into Maiana since the word is our oyster (opensource) and we just can. I imaging this being a custom defined AIS  message. The receiver (Rx) would need to support decoding such message. This should not interfere with receiver that does not support this custom message as well. 

Would appreciate your heads up on this. 
Could we discuss?

**Comment by astuder** [comment]
There are messages types for transmitting unspecified binary messages, some of which are for transmitting weather data:
https://gpsd.gitlab.io/gpsd/AIVDM.html#_type_6_binary_addressed_message
https://gpsd.gitlab.io/gpsd/AIVDM.html#_type_8_binary_broadcast_message

**Reply by astuder** [reply]
See section 10 in this document for details on how weather data is encapsulated in a binary broadcast message:
http://vislab-ccom.unh.edu/~schwehr/papers/2010-IMO-SN.1-Circ.289.pdf

**Reply by revolteur** [reply]
@astuder Thank you for your quick reply and for sharing the resources up top. I will take a look.
